src/test/java/uk/gov/di/ipv/service/ConfigurationServiceTest.java

```diff
@@ -35,11 +41,35 @@ void getCredentialIssuers() throws URISyntaxException {
                                 new CredentialIssuerConfig(
                                         "FraudIssuer",
                                         new URI("http://www.example.com"),
-                                        new URI("http://www.example.com/credential"))));
+                                        new URI("http://www.example.com/credential"))),
+                        CREDENTIAL_ISSUER_CONFIG_BASE64_VERSION_1);
+    }
+
+    @Test
+    void shouldReturnDifferentCredentialIssuersWhenBase64EncodingHasChanged() {
+        ConfigurationService underTest = new ConfigurationService(ssmProvider);
+        when(ssmProvider.get(any()))
+                .thenReturn(
+                        CREDENTIAL_ISSUER_CONFIG_BASE64_VERSION_1,
+                        CREDENTIAL_ISSUER_CONFIG_BASE64_VERSION_2);
 
-        when(ssmProvider.get(any())).thenReturn(CREDENTIAL_ISSUER_CONFIG_BASE64);
+        CredentialIssuers credentialIssuers1 = underTest.getCredentialIssuers(credentialIssuers);
+        CredentialIssuers credentialIssuers2 = underTest.getCredentialIssuers(credentialIssuers);
 
+        assertNotEquals(credentialIssuers1, credentialIssuers2);
+    }
+
+    @Test
+    void shouldReturnSameCredentialIssuersWhenBase64EncodingHasNotChanged() {
         ConfigurationService underTest = new ConfigurationService(ssmProvider);
-        assertEquals(credentialIssuers, underTest.getCredentialIssuers());
+        when(ssmProvider.get(any()))
+                .thenReturn(
+                        CREDENTIAL_ISSUER_CONFIG_BASE64_VERSION_1,
+                        CREDENTIAL_ISSUER_CONFIG_BASE64_VERSION_1);
+
+        CredentialIssuers credentialIssuers1 = underTest.getCredentialIssuers(credentialIssuers);
+        CredentialIssuers credentialIssuers2 = underTest.getCredentialIssuers(credentialIssuers);
+
+        assertEquals(credentialIssuers1, credentialIssuers2);
```

Comment:
I think we need to add an assertion that `CredentialIssuerLoader.loadCredentialIssuers` has only been called once in this scenario. As it stands it's not really demonstrating the caching logic we want.

Reply:
done